core-primitives/test/src/mock/handle_state_mock.rs

```diff
@@ -141,4 +144,48 @@ pub mod tests {
 			updated_state.get(key.encode().as_slice()).expect("value for key should exist");
 		assert_eq!(*inserted_value, value.encode());
 	}
+
+	// this is the same test as for the `GlobalFileStateHandler` to ensure we don't have any effects
+	// from having the state in-memory (as here) vs. in file (`GlobalFileStateHandler`)
+	pub fn ensure_subsequent_state_loads_have_same_hash() {
+		let state_handler = HandleStateMock::default();
+		let shard = ShardIdentifier::default();
+
+		let (lock, _) = state_handler.load_for_mutation(&shard).unwrap();
+		let initial_state = Stf::init_state();
+		let initial_state_hash = hash_of(&initial_state.state);
+		state_handler.write(initial_state, lock, &shard).unwrap();
+
+		let state_loaded = state_handler.load_initialized(&shard).unwrap();
+		let loaded_state_hash = hash_of(&state_loaded.state);
+
+		assert_eq!(initial_state_hash, loaded_state_hash);
+	}
+
+	pub fn ensure_encode_and_encrypt_does_not_affect_state_hash() {
+		let state = Stf::init_state();
+		let initial_state_hash = hash_of(&state.state);
+
+		let encrypted_state = encrypt(&state.state);
+		let decrypted_state: SgxExternalitiesType = decrypt(encrypted_state);
+
+		let decrypted_state_hash = hash_of(&decrypted_state);
+
+		assert_eq!(initial_state_hash, decrypted_state_hash);
+	}
+
+	fn hash_of<T: Encode>(encodable: &T) -> H256 {
+		encodable.using_encoded(blake2_256).into()
+	}
+
+	fn encrypt<T: Encode>(payload: &T) -> Vec<u8> {
+		let encoded_and_encrypted = payload.encode();
+		//encryption_key().encrypt(&mut encoded_and_encrypted).unwrap();
+		encoded_and_encrypted
```

Comment:
encoded or encrypted? I don't think its being encrypted in this function. Same for the decryption function below.

Reply:
ah yes, the naming did not follow all the refactoring steps, good catch 👍

Reply:
So this function is now basically just doing

` payload.encode()`
correct?

So either rename `encoded_and_encrypted` to `encoded` or remove it completely.

Reply:
ahm that's what I've done 😄 no more `encrypt` in the name

Reply:
ah darn it, missed the variable name.. I'm just doing all of these changes too hastily and stressed 😞

Reply:
No worries, that's what PRs are here for I guess :)